### PR TITLE
New geometry support

### LINKED ISSUE #608: New geometry classes
https://github.com/qgis/QGIS/commit/68fe5f579157e81faa79fcea9aa251ae73638c2d
Author: Marco Hugentobler
Date:   Fri May 15 15:41:56 2015 +0200

### PR DESCRIPTION
**Pull request by yjacolin**
* Removing trailing spaces
* Add substitutions string
* Add informations on data format/feature type support
* Add informations on vector editing chapter

Fix #608 (New geometry classes)

### CODE REVIEW

**Comment by yjacolin** [comment]
@DelazJ @SrNetoChan review are warmly welcome :)

**Comment by SrNetoChan** [comment]
@yjacolin I think that there is a few new tools because of this new geometries with new buttons and icons in the digitizing toolbar, I think we should add it also. I haven't tested this so I really don't know how differently it works.

**Comment by yjacolin** [comment]
@SrNetoChan see issues which is linked to #613 also.

**Comment by yjacolin** [comment]
Fix also #632

**Comment by yjacolin** [comment]
Please review

File: source/docs/user_manual/working_with_vector/supported_data.rst [hunk 1]
```diff
@@ -32,6 +32,12 @@ Separated data (CSV). Many of the features available in QGIS work the same,
 regardless of the vector data source. This is by design, and it includes the
 identify, select, labelling and attributes functions.
 
+.. note::
```
Comment (SrNetoChan) [inline source/docs/user_manual/working_with_vector/supported_data.rst]: Not sure if this is already written somewhere but, should we also mention that the support of this Geometry types also depends on the provider supporting it as well. So people don't try to create curves in shapefiles for instance.

**Comment by SrNetoChan** [comment]
LGTM

**Comment by yjacolin** [comment]
I will update the PR in regard of your comment and rebase it will master. Thanks.

**Pull request merged**